statistics/shortest_time_to_get_all_singles.rb

```diff
@@ -14,10 +15,13 @@ def query
         CONCAT('[', person.name, '](https://www.worldcubeassociation.org/persons/', person.id, ')') person_link,
         start_date,
         best
-      FROM Results
-      JOIN Persons person ON person.id = personId and person.subId = 1
+      FROM (
+        -- People who have single for every official event.
+        SELECT personId FROM RanksSingle GROUP BY personId HAVING COUNT(eventId) = #{Events::OFFICIAL.length}
+      ) AS all_events_people
+      JOIN Results result ON result.personId = all_events_people.personId
+      JOIN Persons person ON person.id = result.personId and person.subId = 1
```

Comment:
Basically instead of getting all results (which is really a ton of data), we find WCA IDs of people who have single in all the events instead. Then we get results only for these people.